Elastic Beanstalk Auto Scaling Config

ianwestcott opened the pull request:
#### Summary of changes
**Asana Ticket:** [confirm EB autoscaling settings for dotcom](https://app.asana.com/0/1113545750913664/1121774695362205)

1. Add an Elastic Beanstalk configuration that configures each environment's Auto Scaling Group to determine the health of its instances by using the Elastic Load Balancer's health check, instead of the EC2 status checks. This configuration will apply to each environment on the next deploy post merge, and it won't require re-creating the Auto Scaling Group.
1. Enable CloudWatch metric collection on the Auto Scaling Group to allow for future monitoring of instance health and termination events
1. Remove the Elastic Beanstalk config.yml file, as it's meant for local CLI configuration and not meant to be committed. 

<br>
Assigned to: @ryan-mahoney

ryan-mahoney commented:
@ianwestcott how do I test this? :)

I looked at... seems reasonable, but I don't have a great way of knowing that this will actually do what it is intended to do.

ianwestcott commented:
Sorry, I meant to add testing notes but neglected to do so before hitting the green button. 😅 

I deployed this branch to dotcom-dev-green and confirmed that:

- The deploy was successful
- The environment's auto-scaling group switched from EC2 health checks to ELB health checks
- Metrics for the auto-scaling group were flowing to CloudWatch

Deploying to dotcom-dev should yield the same results. Unfortunately finding the auto-scaling group for each environment is difficult given that Elastic Beanstalk gives everything a cryptic name,  but I can help you find it if you want to verify this change independently.

ianwestcott commented:
As for verifying the behavior of using ELB health checks for the auto-scaling group, that's a little more involved, but I'm open to trying to confirm it. I did manually mark an instance as unhealthy in dotcom-dev-green (by running `aws autoscaling set-instance-health --instance-id i-123abc45d --health-status Unhealthy`, and witnessed that the ASG terminated and replaced it. But that doesn't necessarily confirm the ELB health check behavior. I suppose we could more accurately simulate a failure of the dotcom process on an instance by SSH-ing in and shutting it down, and confirming that the instance is reaped and replaced.
